code/game/objects/items/sport.dm

```diff
@@ -0,0 +1,83 @@
+/obj/item/beach_ball
+	name = "beach ball"
+	icon = 'icons/misc/beach.dmi'
+	desc = "An inflatable ball of fun, enjoyed on many beaches."
+	icon_state = "ball"
+	item_state = "beachball"
+	density = FALSE
+	anchored = FALSE
+	w_class = WEIGHT_CLASS_NORMAL
+	force = 0
+	throwforce = 0
+	throw_speed = 1
+	throw_range = 20
+	flags = CONDUCT
+
+/obj/item/beach_ball/baseball
+	name = "baseball"
+	desc = "Take me out to the ball game."
+	icon = 'icons/obj/basketball.dmi'
+	icon_state = "baseball"
+	item_state = "baseball"
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/beach_ball/dodgeball
+	name = "dodgeball"
+	desc = "Used for playing the most violent and degrading of childhood games. This one is connected to the laser tag armour system."
+	icon = 'icons/obj/basketball.dmi'
+	icon_state = "dodgeball"
+	item_state = "dodgeball"
+	var/suit_types = list(/obj/item/clothing/suit/redtag, /obj/item/clothing/suit/bluetag)
+
+/obj/item/beach_ball/dodgeball/throw_impact(atom/hit_atom)
+	. = ..()
+	var/mob/living/carbon/human/M = hit_atom
+	if(ishuman(hit_atom) && istype(M.wear_suit) && (M.wear_suit.type in suit_types))
```

Comment:
did you mean to just check to see if `M.wear_suit` exists, it should never not be that type unless there's a bug elsewhere

Reply:
Thats what i meant, doing like this is better?
![image](https://user-images.githubusercontent.com/77684085/203784658-2a4c30bb-d632-40e2-8fce-6be8481e726b.png)

Reply:
Yes that should work